spec/rubocop/cop/style/if_unless_modifier_spec.rb

```diff
@@ -4,34 +4,37 @@
 # Note: most of these tests probably belong in the shared context "condition modifier cop"
 ######
 RSpec.describe RuboCop::Cop::Style::IfUnlessModifier, :config do
-  let(:ignore_cop_directives) { true }
-  let(:allow_uri) { true }
-  let(:line_length_config) do
-    {
-      'Enabled' => true,
-      'Max' => 80,
-      'AllowURI' => allow_uri,
-      'IgnoreCopDirectives' => ignore_cop_directives,
-      'URISchemes' => %w[http https]
-    }
-  end
-  let(:other_cops) { { 'Layout/LineLength' => line_length_config } }
+  shared_context 'with LineLength settings' do |enabled: true,
```

Comment:
It's cheating in a way, yes. But also it's just hidden one `let` instead of three that we had before.

Why I chose that instead of a method is that we have to specify `let(:other_cops) { ... }`, in every place, which is somewhat clumsy.

Got an idea though, what do you think of:
```ruby
RSpec.describe RuboCop::Cop::Style::IfUnlessModifier, :config do
  def default_line_length_config
     {
        'Enabled' => true,
        'Max' => 80,
        'AllowURI' => true,
        'IgnoreCopDirectives' => true,
        'URISchemes' => %w[http https]
      }
    end

  def self.with_line_length_config(**overrides)
    let(:other_cops) do
      { 'Layout/LineLength' => default_line_length_config.merge(overrides) }
    end
  end

  context 'when AllowURI is false' do
    with_line_length_config(allow_uri: false)
  
    expect_no_offenses(<<~RUBY)
...
```

I hope they don't burn me as a heretic.

Reply:
I actually went the route of setting `other_cops` in each location that needed it. I thought that ended up being easier to understand with less indirection. In some cases I set it to `{}` in order to unset it, which is what we're actually doing somewhat implicitly in the existing tests.